32-bit build failure

Reported by @iav, and I reproduced the issue. Building bcachefs for 32 bit arm fails currently. I did a little investigation, my findings are below.

```
12345678910arm-linux-gnu-ld: fs/bcachefs/buckets.o: in function `bch2_trans_mark_extent':
buckets.c:(.text+0x2254): undefined reference to `__aeabi_ldivmod'
arm-linux-gnu-ld: fs/bcachefs/buckets.o: in function `bch2_mark_extent':
buckets.c:(.text+0x2ad4): undefined reference to `__aeabi_ldivmod'
arm-linux-gnu-ld: fs/bcachefs/tests.o: in function `btree_perf_test_thread':
tests.c:(.text+0xd04): undefined reference to `__aeabi_uldivmod'
arm-linux-gnu-ld: fs/bcachefs/tests.o: in function `bch2_btree_perf_test':
tests.c:(.text+0x2b08): undefined reference to `__aeabi_uldivmod'
arm-linux-gnu-ld: tests.c:(.text+0x2b30): undefined reference to `__aeabi_uldivmod'
arm-linux-gnu-ld: tests.c:(.text+0x2b48): undefined reference to `__aeabi_uldivmod'
```

Looks to be caused (at least in part) by 64-bit division.

The following resolves the issue in tests.c for me.
```
diff --git a/fs/bcachefs/tests.c b/fs/bcachefs/tests.c
index 4d8d50fd7642..a2686a4c9b6c 100644
--- a/fs/bcachefs/tests.c
+++ b/fs/bcachefs/tests.c
@@ -778,7 +778,7 @@ static int btree_perf_test_thread(void *data)
                wait_event(j->ready_wait, !atomic_read(&j->ready));
        }
 
-       ret = j->fn(j->c, j->nr / j->nr_threads);
+       ret = j->fn(j->c, div_u64(j->nr, j->nr_threads));
        if (ret)
                j->ret = ret;
 
@@ -854,11 +854,11 @@ int bch2_btree_perf_test(struct bch_fs *c, const char *testname,
 
        scnprintf(name_buf, sizeof(name_buf), "%s:", testname);
        bch2_hprint(&PBUF(nr_buf), nr);
-       bch2_hprint(&PBUF(per_sec_buf), nr * NSEC_PER_SEC / time);
+       bch2_hprint(&PBUF(per_sec_buf), div64_u64(nr * NSEC_PER_SEC, time));
        printk(KERN_INFO "%-12s %s with %u threads in %5llu sec, %5llu nsec per iter, %5s per sec\n",
                name_buf, nr_buf, nr_threads,
-               time / NSEC_PER_SEC,
-               time * nr_threads / nr,
+               div_u64(time, NSEC_PER_SEC),
+               time * div64_u64(nr_threads, nr),
                per_sec_buf);
        return j.ret;
 }
```

I haven't figured out the source of the issue in buckets.c, but I've narrowed it down to the call to `bch2_bkey_ptrs_c()`.  The definition of bch_extent_entry differs between 32 and 64 bit, but I'm a bit out of my depth there, so I don't know whether or not that's a red herring.

--- a/fs/bcachefs/buckets.c
+++ b/fs/bcachefs/buckets.c
@@ -666,7 +666,7 @@ static s64 ptr_disk_sectors(s64 sectors, struct extent_ptr_decoded p)
 
 	return p.crc.compression_type &&
 		p.crc.compression_type != BCH_COMPRESSION_TYPE_incompressible
-		? DIV_ROUND_UP(sectors * p.crc.compressed_size,
+		? DIV_ROUND_UP_ULL(sectors * p.crc.compressed_size,
 			       p.crc.uncompressed_size)
 		: sectors;
 }

--- a/fs/bcachefs/tests.c
+++ b/fs/bcachefs/tests.c
@@ -778,7 +778,7 @@ static int btree_perf_test_thread(void *data)
 		wait_event(j->ready_wait, !atomic_read(&j->ready));
 	}
 
-	ret = j->fn(j->c, j->nr / j->nr_threads);
+	ret = j->fn(j->c, div64_u64(j->nr, j->nr_threads));
 	if (ret)
 		j->ret = ret;
 
@@ -854,11 +854,11 @@ int bch2_btree_perf_test(struct bch_fs *c, const char *testname,
 
 	scnprintf(name_buf, sizeof(name_buf), "%s:", testname);
 	bch2_hprint(&PBUF(nr_buf), nr);
-	bch2_hprint(&PBUF(per_sec_buf), nr * NSEC_PER_SEC / time);
+	bch2_hprint(&PBUF(per_sec_buf), div64_u64(nr * NSEC_PER_SEC, time));
 	printk(KERN_INFO "%-12s %s with %u threads in %5llu sec, %5llu nsec per iter, %5s per sec\n",
 		name_buf, nr_buf, nr_threads,
-		time / NSEC_PER_SEC,
-		time * nr_threads / nr,
+		div_u64(time, NSEC_PER_SEC),
+		div_u64(time * nr_threads, nr),
 		per_sec_buf);
 	return j.ret;
 }

